[AMD] Default to hipblaslt in gemm

xw285cornell opened the pull request:
Summary: It has been a constant pain that we have to specify env var to go with the hipblaslt path. The default path is very slow on MI300. Therefore, let's default to hipblaslt.

Differential Revision: D58150764

xw285cornell commented:
@jeffdaily I added ciflow/rocm, wondering if you can help take a look if all the rocm tests are triggered

jeffdaily commented:
> @jeffdaily I added ciflow/rocm, wondering if you can help take a look if all the rocm tests are triggered

Added ciflow/inductor and ciflow/periodic.

aaronenyeshi commented on aten/src/ATen/Context.h:
```diff
@@ -385,8 +385,11 @@ class TORCH_API Context {
-      (c10::utils::check_env("TORCH_BLAS_PREFER_CUBLASLT") == true ||
-       c10::utils::check_env("TORCH_BLAS_PREFER_HIPBLASLT") == true)
+#ifdef USE_ROCM
+      (c10::utils::check_env("TORCH_BLAS_PREFER_HIPBLASLT") != false)
```
Might be cleaner to have both checks the same, either == true or != false.

xw285cornell replied:
yeah it's intentional. The behavior of check_env is that it'll return nullopt when the env var is not set (the most common case), and true if 1, false if 0: https://github.com/pytorch/pytorch/blob/ca561d639b40a9fce088262e2fb35c7dfb61d588/c10/util/env.h#L16. Therefore checking != false makes true as the default, and == true will make false as the default.

aaronenyeshi reviewed (approved):
LGTM

xw285cornell committed:
[AMD] Default to hipblaslt in gemm (#127944)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/127944

It has been a constant pain that we have to specify env var to go with the hipblaslt path. The default path is very slow on MI300. Therefore, let's default to hipblaslt.

Test Plan: OSS CI

Reviewed By: aaronenyeshi, houseroad

Differential Revision: D58150764
